### PR TITLE
add service unavailable and loaded-content pages

### PR DESCRIPTION
**Pull request by VivianAllen**
### What

commit 677b519535de984c6f1d9073027252f97e90caf7 (HEAD -> feat/service-unavailable-page, origin/feat/service-unavailable-page)
Author: Vivian Allen <vivian.allen@methods.co.uk>
Date:   Mon Sep 12 14:45:07 2022 +0100

    Fix typescript syntax errors in categeory Helper tests + types

commit 494ee9afd19aeb16aae935dc54b76bccccbb4949
Author: Vivian Allen <vivian.allen@methods.co.uk>
Date:   Fri Sep 9 15:54:40 2022 +0100

    loaded-content page to show contentStore contents

    - Add content.ts with list of content json files and their
    corresponding data base urls.
    - Rename topicStore -> contentStore and add topic-level array
    listing all loaded releases from content.ts
    - Add new page, loaded-content, that displays the contents of the
    content store
    - remove env vars that load content.json list from json string, as
    this is redundant with addition of content.ts file.

commit df9447be486b0f0d19d177daa60910965bf0dbfa
Author: Vivian Allen <vivian.allen@methods.co.uk>
Date:   Thu Sep 8 15:04:18 2022 +0100

    add SeviceUnavailablePage to render if no topics could be loaded

    - Add new component, ServiceUnavailablePage.svelte, which displays
    a simple header and text explaining the service is unavailable. This
    page will be rendered if no topics could be loaded on startup.

### How to review

To see the service unavailable page, run the atlas locally without any topics with `CONTENT_JSONS='[]' npm run dev` (run the atlas with a blank list of content.json to load). You should see the service unavailable page. If you then run `unset CONTENT_JSONS && npm run dev` (unset the env var so that the content jsons configured at build will be loaded). You should see the atlas work as normal.

To see the loaded-content page, just navigate to `/loaded-content` and you should see a pretty-printed json with the currently loaded content.

### Who can review

Anyone

### CODE REVIEW

File: src/routes/loaded-content/__layout.reset.svelte [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  import "../../i18n/i18n.ts";
+  import Layout from "../../components/Layout.svelte";
+  import ServiceUnavailablePage from "../../components/ServiceUnavailablePage.svelte";
+  import { setContentStoreOnce } from "../../data/setContentStore"
+  import { contentStore } from "../../stores/stores"
+  import { onMount } from 'svelte';
+
+  /* 
+    Most components loaded will need the topicStore to be populated before they will work properly. As this component 
+    loads first, populate the topicStore here! NB - setTopicStoreOnce will only set the topicStore if it is not
```
[the remark this answers is not in the document]
Comment (petem-methods) [reply src/routes/loaded-content/__layout.reset.svelte:13]: (references to TopicStore I mean)

File: src/types.ts [hunk 2]
```diff
@@ -133,3 +134,17 @@ export type DataTileGrid = {
...
+};
+
+export type ContentConfig = {
+  contentBaseUrl: string;
+  contentJsonUrl: string;
+};
+
+export type ContentStore = {
+  releases: [string];
+  topics: [Topic];
```
[the remark this answers is not in the document]
Comment (VivianAllen) [reply src/types.ts:149]: I think 'grouping' doesn't tell you much as a name, though - grouping of what? How about variableGroups?

File: src/types.ts [hunk 3]
```diff
@@ -133,3 +134,17 @@ export type DataTileGrid = {
...
+  envName: string;
+};
+
+export type ContentConfig = {
+  contentBaseUrl: string;
+  contentJsonUrl: string;
+};
+
+export type ContentStore = {
+  releases: [string];
```
[the remark this answers is not in the document]
Comment (VivianAllen) [reply src/types.ts:148]: yeah I realised this mistake only recently - thought I'd got them all but missed this one, will fix

File: src/routes/loaded-content/__layout.reset.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (VivianAllen) [reply src/routes/loaded-content/__layout.reset.svelte:13]: will fix

File: src/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (VivianAllen) [reply src/types.ts:149]: also, there's a new branch for getting rid of topics/topicGroups (its needs the conent scripts refactoring and other stuff): https://github.com/ONSdigital/dp-census-atlas/tree/feat/remove-topics-and-topic-groups-replace-with-census-data-groups

[the remark this answers is not in the document]
Comment (VivianAllen) [reply src/types.ts:149]: So I'd say this rename is OOS for this PR, but can discuss once everyones in?

File: src/routes/loaded-content/__layout.reset.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (VivianAllen) [reply src/routes/loaded-content/__layout.reset.svelte]: done

File: src/types.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (VivianAllen) [reply src/types.ts]: done

**Review by petem-methods — APPROVED** [review]

**Pull request merged**